### PR TITLE
change way of find buttons and fix some issue

### PR DESCRIPTION
**Pull request by lgl017temp**
- change way of find buttons from text match to react data key / index match for resolve not work after translated issue
- fix population not rebalance when no unassignedPopulation
- fix army-explore not woek when neither scout nor explorer added
- disable clicker when option open

### CODE REVIEW

File: src/utils/constants.js [hunk 1]
```diff
@@ -20,6 +30,18 @@ const SUBPAGES = {
...
 
+const SUBPAGES_INDEX = {
+  [SUBPAGES.CITY]: 0,
+  [SUBPAGES.COLONY]: 1,
+  [SUBPAGES.RESEARCH]: 0,
+  [SUBPAGES.SPELLS]: 0,
+  [SUBPAGES.PRAYERS]: 1,
+  [SUBPAGES.ARMY]: 0,
+  [SUBPAGES.EXPLORE]: 1,
+  [SUBPAGES.ATTACK]: 3,
```
Comment (moriakaice) [inline src/utils/constants.js:41]: Would that work if one doesn't have Espionage researched?

File: src/utils/constants.js [hunk 2]
```diff
@@ -8,6 +8,16 @@ const PAGES = {
   DIPLOMACY: 'Diplomacy',
 }
 
+const PAGES_INDEX = {
```
Comment (moriakaice) [inline src/utils/constants.js:11]: The same worry here as with subpages - how does it react when a tab isn't available yet and messes with the count?

**Comment by moriakaice** [comment]
Hey @lgl017temp, I really appreciate the contribution. The way to get the React store is great and opens tons of possibilities (like setting the exact amount when recruiting troops). At the same time, I'm not sure how stable relying on button indexes is, as opposed to looking for the text (that's based on the i18n from the game). Have you tested it when some of the options were unavailable and changed the order?

File: src/utils/constants.js (hunk 2 above)
Comment (lgl017temp) [reply src/utils/constants.js:11]: image

Index is fixed，you can see index 3 is skiped when "Magic" not unlocked

File: src/utils/constants.js (hunk 1 above)
Comment (lgl017temp) [reply src/utils/constants.js:41]: image

**Comment by lgl017temp** [comment]
Emm, Ijust use index at page, sub page and confirm。
Page and sub page index is fixed, if game author not add a new tab at middle (game author use index to mark too)
image

On confirm, Unfortunately not, but I think unless game author consciously revises order, it will not change

looking for the text will not working when translated
you can see a chinese third-party translation version here [https://theresmoregame.g8hh.com/](https://theresmoregame.g8hh.com/)

**Comment by lgl017temp** [comment]
If you are worried about those changes, you can download from my fork and do some test : )

**Comment by moriakaice** [comment]
Thanks for the info! I'll try to review it over the weekend!

**Pull request merged**